classifysql/classify.go

```diff
@@ -0,0 +1,357 @@
+package classifysql
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/muir/sqltoken"
+)
+
+// Dialect enumerates supported SQL dialects.
+type Dialect int
+
+const (
+	DialectMySQL Dialect = iota
```

Comment:
Shouldn't the zero be `DialectGeneric` or `DialectUnknown`?

Reply:
I added a DialectInvalid as the zero.